subsys/bluetooth/audio/ascs.c

```diff
@@ -34,6 +34,8 @@ LOG_MODULE_REGISTER(bt_ascs, CONFIG_BT_ASCS_LOG_LEVEL);
 #include "pacs_internal.h"
 #include "cap_internal.h"
 
+#define ASE_BUF_SEM_TIMEOUT    K_MSEC(50)
```

Comment:
I'm not sure about the timeout chosen, it may cause the code to be unpredictable. Shouldn't the semaphore return immediately when already taken? Where this 50msec value come from?

Reply:
Pulled up from a hat 8)

The idea is that when we generate the read response or notification, and the semaphore is taken, we allow for a short delay to let any other threads that are modifying the ASE to finish. 

The alternative is to reject e.g. the read request immediately if it is being modified, which may also work, but would give a poorer unicast client experience, as their read requests are more likely to be rejected for reasons they cannot control.